gdsfactory/technology/yaml_utils.py

```diff
@@ -44,5 +46,9 @@ def _str_presenter(
             return dumper.represent_scalar("tag:yaml.org,2002:str", data, style="|")
         return dumper.represent_scalar("tag:yaml.org,2002:str", data)
 
-    yaml.add_representer(str, _str_presenter)
-    yaml.representer.SafeRepresenter.add_representer(str, _str_presenter)
+    TechnologyDumper.add_representer(str, _str_presenter)
+
+
+add_color_yaml_representer()
+add_tuple_yaml_representer()
+add_multiline_str_yaml_representer()
```

Comment:
Do we want all of these being run when the file is run, this seems like it could cause more side effects in the future

Reply:
in this case, it's ok because now the functions are only modifying this new `TechnologyDumper` class, which is always imported from this module, so it should not create side effects